testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -1210,39 +1217,48 @@ impl BlockMinerThread {
             e
         })?;
 
-        if block.txs.is_empty() {
+        if block_metadata.block.txs.is_empty() {
             return Err(ChainstateError::NoTransactionsToMine.into());
         }
         let mining_key = self.keychain.get_nakamoto_sk();
         let miner_signature = mining_key
-            .sign(block.header.miner_signature_hash().as_bytes())
+            .sign(
+                block_metadata
+                    .block
+                    .header
+                    .miner_signature_hash()
+                    .as_bytes(),
+            )
             .map_err(NakamotoNodeError::MinerSignatureError)?;
-        block.header.miner_signature = miner_signature;
+        block_metadata.block.header.miner_signature = miner_signature;
 
         info!(
             "Miner: Assembled block #{} for signer set proposal: {}, with {} txs",
```

Comment:
I feel like we should add some tenure budget tracking to this log line. Maybe:

```rust
"tenure_usage" => block_metadata.tenure_budget.proportion_largest_dimension(&block_metadata.tenure_cost),
```

Reply:
We do log that already in stackslib/src/chainstate/nakamoto/miner.rs:649:
```rust
        info!(
            "Miner: mined Nakamoto block";
            "stacks_block_hash" => %block.header.block_hash(),
            "stacks_block_id" => %block.header.block_id(),
            "height" => block.header.chain_length,
            "tx_count" => block.txs.len(),
            "parent_block_id" => %block.header.parent_block_id,
            "block_size" => tenure_size,
            "execution_consumed" => %tenure_consumed,
            "percent_full" => tenure_budget.proportion_largest_dimension(&tenure_consumed),
            "assembly_time_ms" => ts_end.saturating_sub(ts_start),
            "consensus_hash" => %block.header.consensus_hash
        );
```